ports/openrct2/OpenRCT2.sh

```diff
@@ -13,70 +13,55 @@ else
 fi
 
 source $controlfolder/control.txt
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
 if [ -z ${TASKSET+x} ]; then
   source $controlfolder/tasksetter
 fi
-
 get_controls
 
-## TODO: Change to PortMaster/tty when Johnnyonflame merges the changes in,
-CUR_TTY=/dev/tty0
+# Variables
+GAMEDIR="$directory/ports/openrct2"
+ARGS="--user-data-path=save --openrct2-data-path=engine/share/openrct2 --rct2-data-path=RCT2/ --rct1-data-path=RCT1/"
 
-PORTDIR="/$directory/ports"
-GAMEDIR="$PORTDIR/openrct2"
+# CD and set permissions
 cd $GAMEDIR
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+$ESUDO chmod +x -R $GAMEDIR/*
 
-$ESUDO chmod 666 $CUR_TTY
-$ESUDO touch log.txt
-$ESUDO chmod 666 log.txt
-export TERM=linux
-printf "\033c" > $CUR_TTY
-
-printf "\033c" > $CUR_TTY
-## RUN SCRIPT HERE
+# Exports
+export TEXTINPUTPRESET="Name"
+export TEXTINPUTINTERACTIVE="Y"
+export TEXTINPUTNOAUTOCAPITALS="Y"
+export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+export LD_LIBRARY_PATH="$GAMEDIR/libs:$LD_LIBRARY_PATH"
 
-if [ ! -f "RCT2/Data/g1.dat" ]; then
-  echo "Missing game files, see README for more info." > $CUR_TTY
+## Check for game files
+if [ ! -f "$GAMEDIR/RCT2/Data/g1.dat" ]; then
+  pm_message "Missing game files, see README for more info."
   sleep 5
-  printf "\033c" > $CUR_TTY
   exit 1
 fi
 
 # Extract the game if it exists
 if [ -f "$GAMEDIR/engine.zip" ]; then
   if [ -d "$GAMEDIR/engine" ]; then
-    echo "Removing old engine. One moment." > $CUR_TTY
+    pm_message "Removing old engine. One moment."
     $ESUDO rm -fRv "$GAMEDIR/engine"
   fi
 
-  echo "Extracting engine files. Please wait." > $CUR_TTY
   # Extract the engine from the build zip.
+  pm_message "Extracting engine files. Please wait."
   $ESUDO unzip "$GAMEDIR/engine.zip"
   $ESUDO mv -fv "$GAMEDIR/engine/bin/openrct2" "$GAMEDIR/openrct2"
   $ESUDO rm -f "$GAMEDIR/engine.zip"
 fi
 
-export TEXTINPUTPRESET="Name"
-export TEXTINPUTINTERACTIVE="Y"
-export TEXTINPUTNOAUTOCAPITALS="Y"
-export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
-
-export LD_LIBRARY_PATH="$GAMEDIR/libs:$LD_LIBRARY_PATH"
-
-echo "Starting game." > $CUR_TTY
-
+# Run the game
+pm_message "Starting game."
 $GPTOKEYB "openrct2" -c openrct2.gptk textinput &
```

Comment:
if you have gptokeyb env exports you don't need this line.
```suggestion
$GPTOKEYB "openrct2" -c openrct2.gptk &
```